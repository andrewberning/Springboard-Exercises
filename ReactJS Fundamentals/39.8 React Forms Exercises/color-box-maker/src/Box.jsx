import React from 'react';
import './Box.css';

function Box({ 
  id, 
  width = 5, 
  height = 5, 
  color = 'blue', 
  handleRemove 
}) {
  const remove = () => handleRemove(id);

  return (
    <div>
      <div 
      className='Box'
      style={{ 
        width : `${width}em`, 
        height : `${height}em`, 
        backgroundColor: color 
      }}
      data-testid="box">
      </div>
      <button onClick={remove}>X</button>
    </div>
  );
}

export default Box;