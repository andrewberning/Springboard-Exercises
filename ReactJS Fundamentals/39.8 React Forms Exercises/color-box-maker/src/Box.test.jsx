import { expect, it, vi } from 'vitest';
import { render, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom';
import Box from './Box';

it("renders without crashing", function () {
  render(<Box />);
});

it("matches snapshot", function() {
  const {asFragment} = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders box with default props", function() {
  const { getByTestId } = render(<Box id="1" />);

  const box = getByTestId('box');
  expect(box).toBeInTheDocument();
  expect(box).toHaveStyle('width: 5em');
  expect(box).toHaveStyle('height: 5em');
  expect(box).toHaveStyle('background-color: rgb(0, 0, 255)');
})

it('renders box with custom props', () => {
  const { getByTestId } = render(<Box id="1" width={10} height={10} color="red" />);
  
  const box = getByTestId('box'); // Ensure data-testid is set correctly
  expect(box).toHaveStyle('width: 10em');
  expect(box).toHaveStyle('height: 10em');
  expect(box).toHaveStyle('background-color: rgb(255, 0, 0)');
});

it("invokes handleRemove when remove button is clicked", function() {
  const handleRemoveMock = vi.fn();
  const { getByText } = render(<Box id="1" handleRemove={handleRemoveMock} />);
  const removeBtn = getByText("X");
  fireEvent.click(removeBtn);

  expect(handleRemoveMock).toHaveBeenCalledWith('1');
})