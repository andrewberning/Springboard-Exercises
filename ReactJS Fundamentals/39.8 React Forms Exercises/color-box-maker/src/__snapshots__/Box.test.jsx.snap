// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Box"
      data-testid="box"
      style="width: 5em; height: 5em; background-color: blue;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
